deeppavlov/core/commands/train.py

```diff
@@ -109,15 +110,15 @@ def train_model_from_config(config_path: str):
             print(json.dumps(report, ensure_ascii=False))
 
 
-def _test_model(model: Inferable, metrics_functions: List[Tuple[str, Callable]],
+def _test_model(model: Component, metrics_functions: List[Tuple[str, Callable]],
                 dataset: Dataset, batch_size=-1, data_type='valid', start_time=None):
     if start_time is None:
         start_time = time.time()
 
     val_y_true = []
     val_y_predicted = []
     for x, y_true in dataset.batch_generator(batch_size, data_type, shuffle=False):
-        y_predicted = list(model.infer(list(x)))
+        y_predicted = list(model(list(x)))
```

Comment:
If x produced by dataset.batch_generator is a dict or np.array only keys will remain from the dict and np.array will be split

Reply:
x is a batch and batch is a list or a tuple